packages/nuxt/src/pages/utils.ts

```diff
@@ -301,47 +377,99 @@ function prepareRoutes (routes: NuxtPage[], parent?: NuxtPage, names = new Set<s
-        route.children = normalizeRoutes(page.children, metaImports).routes
+        route.children = normalizeRoutes(page.children, metaImports, overrideMeta).routes
       }
 
       // Without a file, we can't use `definePageMeta` to extract route-level meta from the file
       if (!page.file) {
-        for (const key of ['name', 'path', 'meta', 'alias', 'redirect'] as const) {
-          if (page[key]) { route[key] = JSON.stringify(page[key]) }
-        }
         return route
       }
 
       const file = normalize(page.file)
       const metaImportName = genSafeVariableName(filename(file) + hash(file)) + 'Meta'
       metaImports.add(genImport(`${file}?macro=true`, [{ name: 'default', as: metaImportName }]))
 
-      let aliasCode = `${metaImportName}?.alias || []`
-      const alias = toArray(page.alias).filter(Boolean)
-      if (alias.length) {
-        aliasCode = `${JSON.stringify(alias)}.concat(${aliasCode})`
+      const metaRoute: NormalizedRoute = {
+        name: `${metaImportName}?.name ?? ${route.name}`,
+        path: `${metaImportName}?.path ?? ${route.path}`,
+        meta: `${metaImportName} || {}`,
+        alias: `${metaImportName}?.alias || []`,
+        redirect: `${metaImportName}?.redirect`,
+        component: genDynamicImport(file, { interopDefault: true })
       }
 
-      route.name = `${metaImportName}?.name ?? ${page.name ? JSON.stringify(page.name) : 'undefined'}`
-      route.path = `${metaImportName}?.path ?? ${JSON.stringify(page.path)}`
-      route.meta = page.meta && Object.values(page.meta).filter(value => value !== undefined).length ? `{...(${metaImportName} || {}), ...${JSON.stringify(page.meta)}}` : `${metaImportName} || {}`
-      route.alias = aliasCode
-      route.redirect = page.redirect ? JSON.stringify(page.redirect) : `${metaImportName}?.redirect || undefined`
-      route.component = genDynamicImport(file, { interopDefault: true })
+      if (route.children != null) {
+        metaRoute.children = route.children
+      }
 
-      return route
+      if (overrideMeta) {
+        metaRoute.name = `${metaImportName}?.name`
+        metaRoute.path = `${metaImportName}?.path ?? ''`
+
+        // skip and retain fallback if marked dynamic
+        // set to extracted value or fallback if none extracted
+        for (const key of ['name', 'path'] satisfies NormalizedRouteKeys) {
+          if (markedDynamic.has(key)) continue
+          metaRoute[key] = route[key] ?? metaRoute[key]
+        }
+
+        // set to extracted value or delete if none extracted
+        for (const key of ['meta', 'alias', 'redirect'] satisfies NormalizedRouteKeys) {
+          if (markedDynamic.has(key)) continue
+
+          if (route[key] == null) {
+            delete metaRoute[key]
+            continue
+          }
+
+          metaRoute[key] = route[key]
+        }
+      } else {
+        if (route.meta != null) {
+          metaRoute.meta = `{ ...(${metaImportName}) || {}), ...${route.meta} }`
```

Comment:
typo on line 460 (extra ")" after "...(${metaImportName}) || {}"

Reply:
Nice spot. It's already fixed in nightly releases, if you are encountering a bug here.